src/vm/internal/otable.h

```diff
@@ -1,15 +1,50 @@
 #ifndef OTABLE_H
 #define OTABLE_H
 
-/*
- * otable.c
- */
-void init_otable(void);
-void enter_object_hash(struct object_t *);
-void remove_object_hash(struct object_t *);
-void remove_precompiled_hashes(char *);
-struct object_t *lookup_object_hash(const char *);
-int show_otable_status(struct outbuffer_t *, int);
-struct array_t *find_ch_n(const char *s);
+#include<cassert>
+#include<memory>
+#include<string>
+#include<unordered_map>
+#include<vector>
 
+#ifdef TESTING
+struct object_t
```

Comment:
This should be defined in the test file, not here.

Reply:
This one is kind of tricky for now but might be fixable with some refactoring. The problem is otable.cc needs an obname field as the function remove is defined now. This is kind of ugly I think but I want to learn some more about C++11/14/17 before fixing it.  I think I will do it for a second pass after I write the living hash and shared string implementations.